From da0c2131025cdfdc781ca87996721927a3a7f548 Mon Sep 17 00:00:00 2001
From: Colin Cross <ccross@android.com>
Date: Tue, 24 Jul 2012 13:29:12 -0700
Subject: [PATCH 611/944] power: smb347-charger: reinitialize registers after
 usb suspend

Enabling usb suspend mode seems to reset the registers, reprogram them
after changing the usb suspend bit.

Change-Id: I4b805705508f990a56ab7fb99beaaf20cb235f16
Signed-off-by: Colin Cross <ccross@android.com>
---
 drivers/power/smb347-charger.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/power/smb347-charger.c b/drivers/power/smb347-charger.c
index 2f3ffb5..eacff4e 100644
--- a/drivers/power/smb347-charger.c
+++ b/drivers/power/smb347-charger.c
@@ -1017,6 +1017,8 @@ static int smb347_mains_set_property(struct power_supply *psy,
 
 		ret = smb347_write(smb, CMD_A, ret);
 
+		smb347_hw_init(smb);
+
 		smb347_set_writable(smb, false);
 		return 0;
 	case POWER_SUPPLY_PROP_CURRENT_MAX:
-- 
1.8.0

